NuRadioReco/modules/efieldToVoltageConverter.py

```diff
@@ -181,17 +167,24 @@ def run(self, evt, station, det):
                             index_of_refraction = ice.get_refractive_index(1, site)
                         travel_time_shift = geo_utl.get_time_delay_from_direction(sim_station.get_parameter(stnp.zenith),
                             sim_station.get_parameter(stnp.azimuth), antenna_position, index_of_refraction)
-                        start_bin = int(round((electric_field.get_trace_start_time() + cab_delay - times_min.min() + travel_time_shift) / time_resolution))
+                        start_time = electric_field.get_trace_start_time() + cab_delay - times_min.min() + travel_time_shift
+                        start_bin = int(round(start_time / time_resolution))
+                        time_remainder = start_time - start_bin * time_resolution
                     else:
-                        start_bin = int(round((electric_field.get_trace_start_time() + cab_delay - times_min.min()) / time_resolution))
+                        start_time = electric_field.get_trace_start_time() + cab_delay - times_min.min()
+                        start_bin = int(round(start_time / time_resolution))
+                        time_remainder = start_time - start_bin * time_resolution
                     self.logger.debug('channel {}, start time {:.1f} = bin {:d}, ray solution {}'.format(channel_id, electric_field.get_trace_start_time() + cab_delay, start_bin, electric_field[efp.ray_path_type]))
-                    new_trace[:, start_bin:(start_bin + len(trace))] = resampled_efield
+                    new_efield.apply_time_shift(time_remainder)
+                    new_trace[:, start_bin:(start_bin + new_efield.get_number_of_samples())] = new_efield.get_trace()
                 trace_object = NuRadioReco.framework.base_trace.BaseTrace()
                 trace_object.set_trace(new_trace, 1. / time_resolution)
+                trace_object.set_trace_start_time(np.min(times_min) - cab_delay)
```

Comment:
why did you introduce this line? Isn't the logic that we want to have it data like, thus all trace start times after the detector simulation are zero?

Reply:
This is only for the debug plot. The trace_object is only used internally, so this does not affect what the module does to the data. It just makes it easier to compare the times in in trace_object to the ones in the E-Field to make it easier to check if the time shift is applied correctly.

Reply:
Perhaps you can you add a comment to this effect to the code? otherwise I worry it will generate confusion...